metaci/testresults/importer.py

```diff
@@ -121,3 +123,102 @@ def populate_limit_fields(testresult, code_unit):
         testresult.worst_limit_percent = worst_limit_test_percent
         testresult.worst_limit_test = worst_limit_test
         testresult.worst_limit_test_percent = worst_limit_test_percent
+
+def import_robot_test_results(build_flow, path):
+    classes = {}
+    methods = {}
+
+    for result in parse_robot_output(path):
+        class_and_method = '{}.{}'.format(result['suite']['name'], result['name'])
+
+        testclass = classes.get(result['suite']['name'], None)
+        if not testclass:
+            testclass, created = TestClass.objects.get_or_create(
+                name = result['suite']['name'],
+                repo=build_flow.build.repo,
+                test_type='Robot',
+            )
+            classes[result['suite']['name']] = testclass
+
+        method = methods.get(class_and_method, None)
+        if not method:
+            method, created = TestMethod.objects.get_or_create(
+                testclass = testclass, name = result['name']
+            )
+            methods[class_and_method] = method
+
+        testresult = TestResult(
+            build_flow = build_flow,
+            method = method,
+            duration = result['duration'],
+            outcome = result['status'],
+            source_file = result['suite']['file'],
+            robot_xml = result['xml'],
+        )
+        testresult.save()
+
+def parse_robot_output(path):
+    """ Parses a robotframework output.xml file into individual test xml files """
+    
+    tree = ET.parse(path)
```

Comment:
Not a big deal here since the xml is machine-generated, but let's get in the habit of using cumulusci.utils.elementtree_parse_file so that the exception message includes the filename when there's a parse error

Reply:
Done.  I also swapped out usage of ET.parse in BuildFlow.load_junit